deny.toml

```diff
@@ -49,7 +49,8 @@ unsound = "deny"
 # A list of advisory IDs to ignore. Note that ignored advisories will still
 # output a note when they are encountered.
 ignore = [
-    #"RUSTSEC-0000-0000",
+    # Unmaintained dependency error that needs more attention due to nested dependencies
+    "RUSTSEC-2024-0370"
 ]
 # Threshold for security vulnerabilities, any vulnerability with a CVSS score
```

Comment:
This seems a temporary workaround  to unblock the build pipeline .

Reply:
Temporary workarounds are constant workarounds, let's avoid tweaks, it's us and our friends that need to maintain it later, and somebody a year from now will swear a lot because we skip real work.

Reply:
This is a temporary workaround for a non-critical warning that blocks our CI pipeline.  
It needs to be fixed in upstream dependencies. 
I don't have a problem with looking into a fix in dependencies, but there's no reason why this should block PRs on this repository.

Reply:
Consider changing line 41 to:
```
# The lint level for unmaintained crates
unmaintained = "warn"
```
This would also unblock the CI pipeline and wouldn't ignore the problem.

Reply:
To avoid tweaks we need to go through all our dependency tree and update (contribute to) all crates which use `proc-macro-error` to use `proc-macro-error2` as advised. This may take few month and we won't be able to merge any PR until then.
To unblock us we do a temporary fix.
I hope we'll have a non lazy volunteer who will take care about it.

Reply:
@acarbonetto it will never be fixed otherwise.
I did the work of finding the solution, shared what packages need to be replaced, and did the research of pros and cons.
It is not a hard fix, especially after the research already been done, why bypass it?

If we have in our dependencies a package that wasn't maintained for 4 years and didn't even get an answer for three years it is absolutely a critical warning, it's more than critical — it's a security issue, and it's a big red flag for our users.

Tried to collect some data from devs for reasons to choose or replace db client — The number one was security (60+, the sec after is good doc with 40-50 — out of 470 responses). 
I asked follow-up question — how do you check security of a package, most of the answer are about the dependencies — amount, scanning, maintained.

470 is a low number and doesn't say anything deterministic, but it for sure an important issue to take care of.

Reply:
I might be missing something, what is the broken dependency?

Reply:
There are no broken ones. `proc-macro-error` isn't updated for a while, so a new advisory was created recently.
If you run `cargo tree`, you will see the dependency tree, here is a part of it:
```
├── dispose v0.5.0
│   └── dispose-derive v0.4.0 (proc-macro)
│       ├── proc-macro-error v1.0.4           <---- boom
```
To eliminate having `proc-macro-error`, we need to update `dispose-derive`, wait for a release, then bump `dispose-derive` version in `dispose` and get it released too. Nice plan, reliable as a swiss watch!
We have `proc-macro-error` in few places in the tree - multiple crates should be updated.